torch/_inductor/kernel/flex_attention.py

```diff
@@ -188,100 +192,106 @@ def convert_output_node_to_buffer(output):
     l_i = tl.zeros([BLOCK_M], dtype=tl.float32)
     acc = tl.zeros([BLOCK_M, BLOCK_DMODEL], dtype=tl.float32)
 
-    # ~~~~~~~~~~~~~~ fully unmasked blocks ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    # FULL_KV_IDX and FULL_KV_NUM_BLKS are always contiguous.
+    offs_m = q_start * BLOCK_M + tl.arange(0, BLOCK_M)
+
+    # FULL_KV_IDX and KV_NUM_BLKS are always contiguous.
     sparse_hz_offset = sparse_idx_z * SPARSE_H + sparse_idx_h
```

Comment:
Noob Q: Lets say mask fn was const over B and H, could we make a  sparse_kv_idx non-contiguous by broadcasting  the B, H dims

Reply:
also feels safer to just use the stride info